tests/UnifiedSpecTests/Constraint/IsBsonType.php

```diff
@@ -167,6 +168,9 @@ private function doMatches($other): bool
             case 'maxKey':
                 return $other instanceof MaxKeyInterface;
 
+            case 'number':
+                return is_int($other) || (PHP_INT_SIZE == 4 && $other instanceof Int64) || is_float($other) || $other instanceof Decimal128Interface;
```

Comment:
Out of curiosity, is there a reason for limiting the `Int64` check to 32-bit platforms? Technically we won't have an `Int64` instance on 64-bit platforms, but is there a practical reason for the check?

Reply:
This likely goes back to the legacy test format's [DocumentMatchConstraint](https://github.com/mongodb/mongo-php-library/blob/master/tests/SpecTests/DocumentsMatchConstraint.php), from which we derived IsBsonType.

I'll note that the corresponding [Matches constraint prepares 64-bit integers](https://github.com/mongodb/mongo-php-library/blob/ecdacff94528eb4bd27c5a053b34bbb2d457eb40/tests/UnifiedSpecTests/Constraint/Matches.php#L427) such that we should never have an Int64 type on a 64-bit platform; however, that doesn't mean IsBsonType needs to be this strict. This PR seems like a good opportunity to relax that logic, so I'll make the change.

Side note: PHPLIB-1001 only applies the the unified test runner, so DocumentMatchConstraint will continue to not support "number" in its `$$type` operator implementation.